src/definition_reader.cpp

```diff
@@ -477,9 +477,10 @@ void DefinitionReader::handleDefinition(SystemTreeNode &sysTreeNode) {
 
   else {
     if (sysTreeNode.s_parent == OTF2_UNDEFINED_SYSTEM_TREE_NODE) {
-      auto status(m_traceDefs.verifyAbstractParent(sysTreeNode.s_name));
-      if (status == true)
+      bool status(m_traceDefs.requiresAbstractParent(sysTreeNode.s_name));
```

Comment:
From first review > Why the constructor for bool?

Reply:
I will fix it asap.

Reply:
and the answer?

Reply:
Why not:
```
sysTreeNode.s_parent = m_traceDefs.abstractParent(sysTreeNode.s_name);
```

Reply:
In definition_reader.cpp
```proper implementation
 if (sysTreeNode.s_parent == OTF2_UNDEFINED_SYSTEM_TREE_NODE) {
      sysTreeNode.s_parent = m_traceDefs.getAbstractParent(sysTreeNode.s_name);
    }
```
Respectively in definition_store.cpp
```proper implementation
uint32_t DefinitionStore::getAbstractParent(const size_t sysNodeName) {
  if (m_sysTreeNodeDb.size() > 1) {
    if (m_sysTreeNodeDb[1].s_name != sysNodeName) {
      m_sysTreeNodeDb[0].s_newRootparent = true;
      return 0;
    }
  }
  return OTF2_UNDEFINED_SYSTEM_TREE_NODE;
}
```